src/mode/modeHandler.ts

```diff
@@ -466,6 +467,10 @@ export class ModeHandler implements vscode.Disposable {
     this._toBeDisposed.push(disposer);
   }
 
+  /**
+   * Anyone who wants to change the behavior of this method should make sure all selection related test cases pass.
+   * Follow this spec https://gist.github.com/rebornix/d21d1cc060c009d4430d3904030bd4c1 to perform the manual testing.
```

Comment:
These cases are mostly about Mouse selection, AFAIK, we don't have a way to mock mouse right now?